src/main/java/meep/tool/Task.java

```diff
@@ -0,0 +1,384 @@
+package meep.tool;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+
+/**
+ * Abstract base for tasks with Todo, Deadline and Event variants. Provides
+ * parsing helpers,
+ * serialization, and due-date checks.
+ */
+public abstract class Task {
+    private static String inputDtfPattern = "yyyy-MM-dd";
+    private static String outputDtfPattern = "MMM dd yyyy";
+    private static DateTimeFormatter inputDtf = DateTimeFormatter.ofPattern(inputDtfPattern);
+    private static DateTimeFormatter outputDtf = DateTimeFormatter.ofPattern(outputDtfPattern);
+
+    private String description;
+    private boolean done;
```

Comment:
For booleans like these perhaps it would be better to use 'isDone'
```suggestion
    private boolean isDone;
```